spec/models/user_valid_spec.rb

```diff
@@ -0,0 +1,67 @@
+RSpec.describe User do
+  describe '#presenseメソッド' do
+    context 'emailが存在する場合' do
+      let(:user) { build(:user, email: 'hoge@example.com') }
+
+      it 'valid?がtrueになる' do
+        expect(user.valid?).to eq true
+      end
+    end
+
+    context 'emailが空文字の場合' do
+      let(:user) { build(:user, email: '') }
+
+      it 'valid?がfalseになり、errorsに「メールアドレスを入力してください」と格納される' do
+        aggregate_failures do
+          result = user.valid?
+          expect(result).to eq false
+          expect(user.errors.full_messages).to include('メールアドレスを入力してください')
+        end
+      end
+    end
+
+    context 'emailがnilの場合' do
+      let(:user) { build(:user, email: nil) }
+
+      it 'valid?がfalseになり、errorsに「メールアドレスを入力してください」と格納される' do
+        aggregate_failures do
+          result = user.valid?
+          expect(result).to eq false
+          expect(user.errors.full_messages).to include('メールアドレスを入力してください')
+        end
+      end
+    end
+
+    context 'passwordが存在する場合' do
+      let(:user) { build(:user, password: 'hogehoge') }
+
+      it 'valid?がtrueになる' do
+        expect(user.valid?).to eq true
+      end
+    end
+
+    context 'passwordが空文字の場合' do
+      let(:user) { build(:user, password: '') }
+
+      it 'valid?がfalseになり、errorsに「パスワードを入力してください」と格納される' do
+        aggregate_failures do
+          result = user.valid?
+          expect(result).to eq false
+          expect(user.errors.full_messages).to include('パスワードを入力してください')
+        end
+      end
+    end
+
+    context 'passwordがnilの場合' do
+      let(:user) { build(:user, password: nil) }
+
+      it 'valid?がfalseになり、errorsに「パスワードを入力してください」と格納される' do
+        aggregate_failures do
+          result = user.valid?
+          expect(result).to eq false
+          expect(user.errors.full_messages).to include('パスワードを入力してください')
```

Comment:
以下の方がより信頼性のあるテストになるよ〜

includeの場合は、`'パスワードを入力してください'`という文字列が要素に含まれる、ということだけテストしているので、`user.errors.full_messages`に他のメッセージが含まれていてもテストが通ってしまうんよね。
```suggestion
          expect(user.errors.full_messages).to eq ['パスワードを入力してください']
```

Reply:
配列のまま渡して判定に使えるのかー

下記コミットで修正しました
965c1b71fc314f513a280ced1eae719e30059d20

Reply:
@Kaito1290 
eqだと使えるね！